server/server.go

```diff
@@ -1049,12 +1065,18 @@ func (s *server) GetSnapshotNames(ctx context.Context, req *rpcpb.GetSnapshotNam
 }
 
 func (s *server) getClusterInfo() *rpcpb.ClusterInfo {
-	s.mu.RLock()
+	s.infoMu.RLock()
 	info := s.clusterInfo
-	s.mu.RUnlock()
+	s.infoMu.RUnlock()
 	return info
 }
 
+func (s *server) setClusterInfo(info *rpcpb.ClusterInfo) {
+	s.infoMu.RLock()
```

Comment:
In my opinion, if at all, this should be a write lock I think.

What I am understanding here is that you are trying to lock the clusterInfo, but you can't use `s.mu` because that would cause a deadlock. Therefore you introduce a new lock, but that one is going to never lock for writing as far as I can see.

I think it's safe to remove this lock altogether. If I have seen this correctly, every method calling `setClusterInfo` is locked via `s.mu` anyway. Therefore there can't be any write conflict. If this assumption is correct, we might just **assume** that the lock is held here (in this case add the comment).

Otherwise, explain what's going on and why this is needed.